### PR TITLE
Add vector and matrix utility functions

### PR DESCRIPTION
**Pull request by axeljaeger**
Utility functions as discussed in:
https://forum.babylonjs.com/t/matrix-vector-color-conversation-and-utilities-cleanup/49818

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Maths/math.vector.functions.ts [hunk 1]
```diff
@@ -29,3 +32,100 @@ export function Vector3ToFixed(vector: Vector3, decimalCount: number): string {
...
+ * directly to a FloatArray at the given offset.
+ * Example Playground - https://playground.babylonjs.com/#AV9X17#25
+ * @param scale defines the scale vector3
+ * @param rotation defines the rotation quaternion
+ * @param translation defines the translation vector3
+ * @param array defines the array to write to
+ * @param offset defines the offset in the target array where to start storing values
+ */
+export function ComposeMatrixToArray<T extends Matrix>(scale: DeepImmutable<Vector3>, rotation: DeepImmutable<Quaternion>, translation: DeepImmutable<Vector3>, array: FloatArray, offset: number): void {
+    const x = rotation._x,
```
Comment (deltakosh) [inline packages/dev/core/src/Maths/math.vector.functions.ts:91]: Isn't it a code duplication of what we already have?

Comment (axeljaeger) [reply packages/dev/core/src/Maths/math.vector.functions.ts:91]: Yes as mentioned in the forum, this is code duplication of `Matrix.ComposeToRef`, unfortunately I don't know how to reuse the existing code. Maybe `ComposeToRef` could make use of this version, but then you would have the `+ offset` with an offset of 0.

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15101/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15101/merge/testResults/webgl2playwright/index.html

**Review by sebavan — COMMENTED** [review]
I am not to sure what those new functions bring ?

Is it just a coding style preference ?

**Comment by axeljaeger** [comment]
Hi Sebavan,
the original idea was to add some functions that are not yet in the library, such as `Vector2.toVector3`, that feel missing because there is already a similar function, such as `Vector3.toVector2`.


As [Deltakosh](https://forum.babylonjs.com/u/Deltakosh) mentioned in the [discussion](https://forum.babylonjs.com/t/matrix-vector-color-conversation-and-utilities-cleanup/49818), for the sake of tree-shaking, we should actually not put more functions in Vector and Matrix, but rather add them to the mentioned utility functions file, so I prepared this MR as a base of discussion.

Actually, personally I got unsure wether it is worth filling this file utility file already with the missing functions while the majority of this functions remains for now in Vector an Matrix classes respectively.

When it is about making the functions pure functions and not part of the classes I can also easily add them somewhere in my project.

What do you think?

**Comment by sebavan** [comment]
> Hi Sebavan, the original idea was to add some functions that are not yet in the library, such as `Vector2.toVector3`, that feel missing because there is already a similar function, such as `Vector3.toVector2`.
> 
> As [Deltakosh](https://forum.babylonjs.com/u/Deltakosh) mentioned in the [discussion](https://forum.babylonjs.com/t/matrix-vector-color-conversation-and-utilities-cleanup/49818), for the sake of tree-shaking, we should actually not put more functions in Vector and Matrix, but rather add them to the mentioned utility functions file, so I prepared this MR as a base of discussion.
> 
> Actually, personally I got unsure wether it is worth filling this file utility file already with the missing functions while the majority of this functions remains for now in Vector an Matrix classes respectively.
> 
> When it is about making the functions pure functions and not part of the classes I can also easily add them somewhere in my project.
> 
> What do you think?

I completely agree that new addition should be in this file and this format, but here I am more questioning the general usefulness of those shortcuts and how frequently we would use them to understand if it justifies the extra maintenance and weight in the UMD scenario :-) 

@deltakosh @RaananW any thoughts?

**Comment by sebavan** [comment]
Lets keep them in your project for now and reopen if we change our mind then ?

**Pull request closed without merging**